internal/controller/apiextensions/claim/configurator.go

```diff
@@ -74,20 +85,20 @@ func Configure(_ context.Context, cm resource.CompositeClaim, cp resource.Compos
 		return nil
 	}
 
-	// TODO(negz): Consider nesting all user-specified spec values under a
-	// predictable object like spec.parameters so we can propagate _only_ user
-	// specified fields. Maintaining a set of keys to delete here seems error
-	// prone. Note that deleting these keys isn't always necessary; the CRD will
-	// drop them if preserveUnknownFields is false, but that is not the default
-	// for pre-v1 CRDs.
-	i, _ := fieldpath.Pave(ucm.Object).GetValue("spec")
-	spec, ok := i.(map[string]interface{})
+	icmSpec := ucm.Object["spec"]
+	spec, ok := icmSpec.(map[string]interface{})
 	if !ok {
 		return errors.New(errUnsupportedClaimSpec)
 	}
 
-	_ = fieldpath.Pave(ucp.Object).SetValue("spec", filter(spec, xcrd.FilterClaimSpecProps...))
+	// Delete base claim fields when configuring composite spec
+	baseClaimSpec := xcrd.CompositeResourceClaimSpecProps()
+	// keep composition selection fields
+	delete(baseClaimSpec, "compositionRef")
+	delete(baseClaimSpec, "compositionSelector")
```

Comment:
What do you think about using something like `xcrd.KeepClaimSpecProps := []string{"compositionRef", "compositionSelector"}`? This might help with readability a bit, and keep the set of fields that we want to keep / propagate close to where the schemas are defined in the xcrd package, similar to the previous `xcrd.FilterClaimSpecProps`.